packages/web-extension/src/common/utils/GeneralUtils.ts

```diff
@@ -19,37 +19,43 @@ export default class GeneralUtils {
   public static nextTimestamp<T extends Api.AmbiguousTimestamp>(
     currentTime: number,
     timestamps: T[],
-    preferences: Api.Preferences | undefined
+    preferences: Api.Preferences | undefined,
+    ignoreAutoSkipPreference?: boolean | undefined
   ): T | undefined {
     if (!preferences) {
       return timestamps.find(timestamp => timestamp.at > currentTime);
     }
     return timestamps.find(
-      timestamp => timestamp.at > currentTime && !GeneralUtils.isSkipped(timestamp, preferences)
+      timestamp =>
+        timestamp.at > currentTime &&
+        !GeneralUtils.isSkipped(timestamp, preferences, ignoreAutoSkipPreference)
     );
   }
 
   public static previousTimestamp<T extends Api.AmbiguousTimestamp>(
     time: number,
     timestamps: T[],
-    preferences: Api.Preferences | undefined
+    preferences: Api.Preferences | undefined,
+    offset = 0.1
   ): T | undefined {
     if (!preferences) {
-      return timestamps.filter(timestamp => timestamp.at < time - 0.1).pop();
+      return timestamps.filter(timestamp => timestamp.at < time - offset).pop();
     }
     return timestamps
       .filter(
-        timestamp => !GeneralUtils.isSkipped(timestamp, preferences) && timestamp.at < time - 0.1
+        timestamp => !GeneralUtils.isSkipped(timestamp, preferences) && timestamp.at < time - offset
       )
       .pop();
   }
 
   public static isSkipped(
     { typeId: _typeId }: Api.AmbiguousTimestamp,
-    preferences: Api.Preferences | undefined
+    preferences: Api.Preferences | undefined,
+    ignoreAutoSkipPreference = false
   ): boolean {
     if (!preferences) return false;
-    if (!preferences.enableAutoSkip) return false;
+    if (!ignoreAutoSkipPreference && !preferences.enableAutoSkip) return false;
+
```

Comment:
Looking at this now, this is a poorly written function, bad job me from 3 years ago 😂 

Lets refactor all these 3 functions (`isSkipped`, `nextTimestamp`, and `prevTimestamp`) so that the if statement checks are done before calling the util functions. That way we don't need to keep adding parameters for each place this is used.

Reply:
Hmm I am not 100 percent sure on how you want the refactor to look, and given that these functions are used all over the place, would you mind doing this refactor?